src/components/steps/nameAndDescription/nameAndDescription.tsx

```diff
@@ -12,10 +14,54 @@ const defaultNameAndDescription = {
   name: '',
 };
 
-const NameAndDescription = () => {
+export type StepsProps = {
+  setSubmitStep: React.Dispatch<
+    React.SetStateAction<{
+      initialized: boolean;
+      submitted: boolean;
+    }>
+  >;
+
+  submitStep: {
+    initialized: boolean;
+    submitted: boolean;
+  };
+};
+
+const NameAndDescription: React.FC<StepsProps> = ({
+  submitStep,
+  setSubmitStep,
+}) => {
   const [nameAndDescription, setNameAndDescription] =
     useState<NameAndDescriptionState>(defaultNameAndDescription);
 
+  const { recipeToUpload, setRecipeToUpload, setDisplayMessage } =
+    useContext(StorageContext);
+
+  useEffect(() => {
+    if (recipeToUpload?.description && recipeToUpload.name) {
```

Comment:
Why don't you destructure the recipeToUpload in the first place?
const { description, name } = recipeToUpload || {};
if ( description && name ) {}